### ISSUE TITLE
Crash when reading / refreshing Vault Generic Secret resources.

### ISSUE THREAD

**Issue by forstermatth** [terraform-providers/terraform-provider-vault#50]
### Terraform Version
Terraform v0.11.0
+ provider.vault v1.0.0

### Vault Version

Vault v0.9.0 ('bdac1854478538052ba5b7ec9a9ec688d35a3335')

### Affected Resource(s)

* `vault_generic_secret`

### Terraform Configuration Files
```hcl

variable "redis_internal_cache_url" { default = "url" }
variable "environment" { default = "staging" }

resource "vault_generic_secret" "redis_cache" {
  path = "${var.environment}/secrets/redis/cache"

  data_json = <<JSON
{
  "url": "${var.redis_internal_cache_url}",
}
JSON

}
```

### Panic Output
Relevant crash log output: https://gist.github.com/forstermatth/9ca3d0d057691d8a7233e2e628d2cbac

### Expected Behavior
Should read and refresh the state.

### Actual Behavior
Tries to read a value, gets a nil pointer and dies.

### Steps to Reproduce
1. `terraform apply` or `terraform refresh` or `terraform plan`

### Important Factoids

* It is trying to refresh ~100 `vault_generic_secrets` and a few `vault_mounts`
* This results in state corruption, and anything to do with refreshing state fails afterwards
* The only way I've figured out how to get around this is to completely wipe the state and start again
* It seems to be inconsistent in which secret it fails on, the null pointer is returned on a different resource often
* The secrets exist already in vault (the plan has been applied previously)
* I am using the Postgres storage back-end

**Comment by jgiles** [comment]
Is this the same as #55 ?

**Comment by forstermatth** [comment]
@jgiles The panic output looks very similar.

**Comment by paultyng** [comment]
I think this will be fixed by a9ddc4775c25695ccb82b3bd2029480bec32247b
